src/test-monitor/test-monitor.cc

```diff
@@ -309,6 +353,12 @@ int main(int argc, char* argv[]) {
     fprintf(stderr, "Invalid timeout %s\n", argv[1]);
     return 1;
   }
+  if (char *diagnose_timeout_str = getenv("RR_TEST_DIAGNOSE_TIMEOUT")) {
+      diagnose_timeout = strtoul(diagnose_timeout_str, &endp, 10);
+      if (*endp) {
+          diagnose_timeout = (unsigned long)-1;
```

Comment:
`ULONG_MAX`?

Reply:
Done. (these are some of the standard macros that I somehow have never got used to using....)